android/walletlib/src/main/java/com/solana/mobilewalletadapter/walletlib/authorization/AccountRecordsDao.java

```diff
@@ -0,0 +1,99 @@
+        implements AccountRecordsDaoInterface, AccountRecordsSchema {
+
+    public AccountRecordsDao(SQLiteDatabase db) { super(db); }
+
+    @NonNull
+    @Override
+    protected AccountRecord cursorToEntity(@NonNull Cursor cursor) {
+        final int publicKeyId = cursor.getInt(cursor.getColumnIndexOrThrow(COLUMN_ACCOUNTS_ID));
+        final byte[] publicKey = cursor.getBlob(cursor.getColumnIndexOrThrow(COLUMN_ACCOUNTS_PUBLIC_KEY_RAW));
+        final String accountLabel = cursor.getString(cursor.getColumnIndexOrThrow(COLUMN_ACCOUNTS_LABEL));
+        final String accountIconStr = cursor.getString(cursor.getColumnIndexOrThrow(COLUMN_ACCOUNTS_ICON));
+        final String chainsString = cursor.getString(cursor.getColumnIndexOrThrow(COLUMN_ACCOUNTS_CHAINS));
+        final String featuresString = cursor.getString(cursor.getColumnIndexOrThrow(COLUMN_ACCOUNTS_FEATURES));
+        final Uri accountIcon = Uri.parse(accountIconStr);
+        final String[] chains = deserialize(chainsString);
+        final String[] features = deserialize(featuresString);
+        return new AccountRecord(publicKeyId, publicKey, accountLabel, accountIcon, chains, features);
+    }
+
+    @Override
+    public long insert(@NonNull byte[] publicKey,
+                       @Nullable String accountLabel,
+                       @Nullable Uri accountIcon,
+                       @Nullable String[] chains,
+                       @Nullable String[] features) {
+        final ContentValues accountContentValues = new ContentValues(4);
+        accountContentValues.put(COLUMN_ACCOUNTS_PUBLIC_KEY_RAW, publicKey);
+        accountContentValues.put(COLUMN_ACCOUNTS_LABEL, accountLabel);
+        accountContentValues.put(COLUMN_ACCOUNTS_ICON, accountIcon != null ? accountIcon.toString() : null);
+        accountContentValues.put(COLUMN_ACCOUNTS_CHAINS, chains != null ? serialize(chains) : null);
+        accountContentValues.put(COLUMN_ACCOUNTS_FEATURES, features != null ? serialize(features) : null);
+        return super.insert(TABLE_ACCOUNTS, accountContentValues);
+    }
+
+    @Nullable
+    @Override
+    public AccountRecord query(@NonNull byte[] publicKey) {
+        final SQLiteDatabase.CursorFactory accountCursorFactory = (db1, masterQuery, editTable, query) -> {
+            query.bindBlob(1, publicKey);
+            return new SQLiteCursor(masterQuery, editTable, query);
+        };
+        try (final Cursor cursor = super.queryWithFactory(accountCursorFactory,
+                TABLE_ACCOUNTS,
+                ACCOUNTS_COLUMNS,
+                COLUMN_ACCOUNTS_PUBLIC_KEY_RAW + "=?",
+                null)) {
+            if (!cursor.moveToNext()) {
+                return null;
+            }
+            return cursorToEntity(cursor);
+        }
+    }
+
+    @Override
+    public void deleteUnreferencedAccounts() {
+        final SQLiteStatement deleteUnreferencedPublicKeys = super.compileStatement(
```

Comment:
`deleteUnreferencedPublicKeys` should be renamed to `deleteUnreferencedAccounts`?

Reply:
done